pkg/handlers/sdk.go

```diff
@@ -22,10 +23,40 @@ type Handler struct {
 }
 
 // Creates a new handler wrapping the SDK, which is authenticated through the cached client-credentials flow tokens
-func New(platformEndpoint, clientID, clientSecret string) (Handler, error) {
+func New(platformEndpoint, clientID, clientSecret string, insecure bool) (Handler, error) {
 	scopes := []string{"email"}
 
-	sdk, err := sdk.New(platformEndpoint, sdk.WithClientCredentials(clientID, clientSecret, scopes), sdk.WithTokenEndpoint(TOKEN_URL), sdk.WithInsecureConn())
+	opts := []sdk.Option{
+		sdk.WithClientCredentials(clientID, clientSecret, scopes),
+		sdk.WithTokenEndpoint(TOKEN_URL),
+	}
+
+	// Try an parse scheme out of platformEndpoint
+	// If it fails, use the default scheme of https
+	// There has to be a better way to do this
+	platformURL, err := url.Parse(platformEndpoint)
+	if err != nil {
+		return Handler{}, err
+	}
+
+	switch platformURL.Scheme {
+	case "http":
+		opts = append(opts, sdk.WithInsecurePlaintextConn())
+		if platformURL.Port() == "" {
+			platformURL.Host += ":80"
+		}
+	case "https":
+		if platformURL.Port() == "" {
+			platformURL.Host += ":443"
+		}
+	default:
+		return Handler{}, errors.New("invalid scheme")
+	}
```

Comment:
@jrschumacher I don't like this approach.  Maybe we need a `--plaintext` flag instead as well?

Reply:
I think this is fine. This is the spec... if you're using it on HTTP you have bigger problems but those problems exist in the platforms.